app/scripts/lib/metrics.js

```diff
@@ -186,6 +194,9 @@ define(function (require, exports, module) {
           }
```

Comment:
> What I worry about with the semaphorish approach is that we'll lose events.

Sorry, I didn't mean to propose this as the actual fix, I just wanted to draw your attention to the problem. Losing events is what I was trying to describe in my point `2` above.

> I wonder if there's any harm calling `clear` before `this._send` and not worrying about setting an `_isFlushing`.

I'm already working on an approach that does this locally. I'll close this PR to prevent any further confusion, sorry about that.